### PR TITLE
chore(deps): migrate webpack 4 to 5

### PR DESCRIPTION
**Pull request by kb-0311**
Updates dev deps ->

```
"html-webpack-plugin" : "^5.50.0" ,
"webpack": "^5.70.0",
"webpack-cli": "^4.9.2"
```

added dependency -`@"babel/runtime" : "^7.17.7"` refer reason ->https://github.com/babel/babel/issues/8462

### CODE REVIEW

**Comment by saghul** [comment]
Doesn't this require changes to our webpack file? I doubt it...

**Comment by kb-0311** [comment]
I can't think of the changes to be applied on `webpack.main.js` or ` webpack.rendered.js` on top of my head .

But if there are changes to be requested, you can suggest some and I will apply it : )

**Comment by saghul** [comment]
Have you actually tested it? As you can see in the CI fails on all platforms?

**Comment by kb-0311** [comment]
ig the errors were resolved for by installing the `@babel/runtime` dep and updating the `html-webpack-plugin` dev dep ... a few warnings still persist though related to atlaskit .

**Comment by kb-0311** [comment]
Nevermind, for some reason after switching to ubuntu those `@atlaskit `errors are gone.

**Comment by kb-0311** [comment]
could you please run the tests again on the three platforms again?

**Comment by kb-0311** [comment]
Great! It passed all tests, could you please merge it ? ... so that I can work on other updates to libraries in other PRs?

Or can you suggest some changes to be made before going for a merge?

**Comment by saghul** [comment]
I want to give it a manual test before, our CI just builds it.

**Comment by kb-0311** [comment]
Yes, for sure please let me know the result when that happens.

On a side note, may I please ask why have we not added and compartmentalized unit tests, E2E tests, Linter tests etc, and other tests for this particular project?

**Comment by csett86** [comment]
Because this project is a hobby project done in spare time, so any contributions of the above mentioned are welcome

**Comment by kb-0311** [comment]
> Because this project is a hobby project done in spare time, so any contributions of the above mentioned are welcome

Nice. I guess after this one is merged (if there are not any issues during manual test) . I will be spending some time on the compatibility side of vulnerable dependencies and then I think I can then start with a few unit test or linter tests.

File: package.json [hunk 1]
```diff
@@ -129,6 +129,7 @@
   "readmeFilename": "README.md",
   "license": "Apache-2.0",
   "dependencies": {
+    "@babel/runtime": "^7.17.7",
```
Comment (csett86) [inline package.json:132]: This should not be part of the dependencies, but instead dev-dependencies, as its packaged and included via webpack. Please update.

**Comment by csett86** [comment]
Not sure how you tested it, @kb-0311, but running 

    npm ci
    npm run dist

results in a non-functional app:



Please test and fix your PR (and incorporate the other comment on the dep -> dev-dep). Thank you

**Comment by kb-0311** [comment]
@csett86  That's weird, I am not getting
Screenshot from 2022-03-21 07-10-32
 those errors on running `npm run dist` on my machine. 

Screenshot from 2022-03-21 07-10-14

**Comment by kb-0311** [comment]
Also in the CI test the dist command is checked for - whether it results in a proper built or not (You can see in the details :+1: ).  That test case was passed,

May I know what OS you are using? maybe your local machine is incompatible with some library installation ?

**Comment by csett86** [comment]
@kb-0311 The issue is not running the npm run dist command, the issue is with the resulting application. That is broken with your PR.

You can check yourself by downloading the binary for the platform you have available at the last github action run for this PR: https://github.com/jitsi/jitsi-meet-electron/actions/runs/1999644241

When launching, it fails with the same error that I pasted above. To make it visible under linux, run it like:

    SHOW_DEV_TOOLS=true ./jitsi-meet-x86_64.AppImage

**Comment by kb-0311** [comment]
@csett86  Ohhhh now I get what you mean. Sorry for misunderstanding earlier and Thank you for your patience on this one :+1: 

I think the issue here is the webpack v5 does not auto-add polyfill for process variables and that needs [to be added manually ](https://github.com/vfile/vfile/issues/38#issuecomment-640479137).

but doing that results in a different error -> 'browser' not defined. 

May I please ask how are the process variables currently handled?

**Comment by csett86** [comment]
@kb-0311 Please see the webpack migration guide on what has changed to v5 and how to handle it: https://webpack.js.org/migrate/5/

**Comment by kb-0311** [comment]
@csett86 Thanks for the guide. Could you please check out the last commit ? It seem to be working well on my machine.
Screenshot from 2022-03-22 16-03-39

**Comment by kb-0311** [comment]
could you please test this one again ?

**Comment by csett86** [comment]
@kb-0311, please test all of the app functions yourself first (eg. video, always-on-top window, screensharing, settings), on all platforms, ie windows, linux and mac. Thank you

**Comment by kb-0311** [comment]
> @kb-0311, please test all of the app functions yourself first (eg. video, always-on-top window, screensharing, settings), on all platforms, ie windows, linux and mac. Thank you

Yes.Seems to be okay.

### Starting
Screenshot from 2022-03-23 12-03-51


### Join A Meet
videoStart

### Performance Setting
peroformance

 ### Audio Taking in Input
micrrophone taking input
 
###  Raised Hand
raisedHand

### Chats
ChatsandPolls

### Audit log
auditLog
### Screen sharing the Audit Log
ScreenSharing AuditLog
### Ending the meet 
endingTHeMeet

**Comment by csett86** [comment]
Thank you for testing, @kb-0311 , I have two more remarks from my testing:

- npm run start fails because webpack 5 no longer recognises `--watch-poll`, please fix, the upgrade guide should help you with that:

```
[watch] [webpack-cli] Error: Unknown option '--watch-poll'
[1] 21:17:46.755 › Skip checkForUpdatesAndNotify because application is not packed
[watch] [webpack-cli] Run 'webpack --help' to see available commands and options
[watch] npm run watch exited with code 2
```

- @babel/runtime is not required as a dependency, I have tested to remove it, and the build continues to work (as the devDependencies that require it anyway already it as a dependency). Please update as well.

Currently the CI seems to hang (I already approved it yesterday), so please just push the above updated, then I will approve it again.

**Comment by kb-0311** [comment]
@csett86 Thanks for your help on this one! I think those errors must be gone now. Plus I added `@babel/runtime` as a dev dependency.

Also for `webpack v5.7.0 ` made this change that you requested.

`--watch-poll` -> `--watch-options-poll=1000`

Let me know further suggestions if any. Thank you.

**Comment by csett86** [comment]
@kb-0311 Two more things:

- please remove @babel/runtime, its also not required in the devDependencies
- Please fix the linter errors (run npm run lint locally to see them)

Regarding rebase: No, dont rebase, as there is no conflict with the new commits its not necessary

**Comment by kb-0311** [comment]
@csett86 resolved most of the linter errs but this one is still remaining and I don't know how to solve it. removed `@babel/runtime`

**Comment by csett86** [comment]
This seems to be newly added by the dependency updates you did. As this file is part of the tests of the dependency, you need to exclude it from flow via

https://github.com/jitsi/jitsi-meet-electron/blob/master/.flowconfig

**Comment by kb-0311** [comment]
@csett86 Thanks for your prompt and quick answer. I don't think there are any more linter errors.

**Comment by csett86** [comment]
Now as everything seems settled, can you please run another round of testing the core functionalities, please @kb-0311

**Comment by kb-0311** [comment]
@csett86  Yes for sure :+1: . Everything seems to be running well.

### Starting a meet
startingAmeet
### Microphone taking Input
audioMicTakingInput
### Raised Hand
raisedHand
### Chats
chats
### Voting for a Poll .
votingInAPoll
### Results of the poll 
resultsOfAPoll
### Screen Sharing Audit Log
ScreenSharingTheAuditLog
### Members
members
### Playing Around with Background
differentBackground

**Comment by csett86** [comment]
Thank you, would you be able to test on windows an mac once as well?

**Review by csett86 — APPROVED** [review]
I have tested on windows 10 and macOS 10.14.6, looks good there as well. Will merge now.

Thank you for your contribution, @kb-0311, looking forward to the next one :)

**Pull request merged**